app/models/agents/webhook_agent.rb

```diff
@@ -38,10 +41,14 @@ def default_options
     end
 
     def receive_web_request(params, method, format)
+      # check the secret
       secret = params.delete('secret')
-      return ["Please use POST requests only", 401] unless method == "post"
       return ["Not Authorized", 401] unless secret == interpolated['secret']
 
+      #check the verbs
+      verbs = (options['verbs'] || 'post').split(/[,;]/).map { |x| x.strip.downcase }
```

Comment:
Using `interpolated['verbs'].presence || 'post'` would allow someone to write Liquid if they really wanted to.